app/uk/gov/hmrc/formpproxy/sdlt/repositories/SdltFormpRepository.scala

```diff
@@ -933,4 +935,31 @@ class SdltFormpRepository @Inject() (@NamedDatabase("sdlt") db: Database)(implic
     } finally cs.close()
   }
 
+  override def sdltDeletePredefinedAgent(request: DeletePredefinedAgentRequest): Future[DeletePredefinedAgentReturn] =
+    Future {
+      db.withTransaction { conn =>
+        callDeleteAgent(
+          conn = conn,
+          p_storn = request.storn,
+          p_agent_resource_ref = request.agentReferenceNumber
+        )
+      }
+    }
+
+  private def callDeleteAgent(
+    conn: Connection,
+    p_storn: String,
+    p_agent_resource_ref: String
```

Comment:
change:
`    p_agent_resource_ref: Long
`
it's defined as a number [here](https://github.com/hmrc/prh-oracle-xe/blob/main/databases/sdlt/scripts/tables.tab#L21C2-L21C36)